### ISSUE TITLE
Layout/RescueEnsureAlignment doesn't behave as expected with inline private_class_method

### ISSUE THREAD

**Issue by dgmora** [rubocop-hq/rubocop#7170]
Rubocop reports a warning, when I think it shouldn't.

--------

## Expected behavior
I expect no warnings for this example. Also, autocorrection should align this method as it is:
```ruby
private_class_method def self.parse(_foo)
  puts 'hi'
rescue StandardError
  puts 'do nothing'
end
```

## Actual behavior

This is the warning I get:
```
Inspecting 1 file
C

Offenses:

tests.rb:5:1: C: Layout/RescueEnsureAlignment: rescue at 5, 0 is not aligned with def self.parse at 3, 21.
rescue StandardError
^^^^^^

1 file inspected, 1 offense detected
```
And this the autocorrection:
```ruby
# frozen_string_literal: true

private_class_method def self.parse(_foo)
  puts 'hi'
                     rescue StandardError
                       puts 'do nothing'
end
```

## Steps to reproduce the problem

Already described

## RuboCop version

```
0.71.0 (using Parser 2.6.3.0, running on ruby 2.5.3 x86_64-darwin17)
```
